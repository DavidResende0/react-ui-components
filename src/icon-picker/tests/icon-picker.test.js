import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Modal } from 'patternfly-react';

import IconPicker from '../icon-picker';
import FontIcon from '../font-icon';
import IconsList from '../icon-list';

describe('IconList test', () => {
  let intialProps;
  let styleTag;
  beforeEach(() => {
    intialProps = {
      iconTypes: [{ type: 'fa', title: 'Font Awesome' }, { type: 'pf', title: 'PatternFly' }],
    };
  });

  beforeAll(() => {
    styleTag = document.createElement('style');
    styleTag.innerHTML = `
      .fa-crosshairs::before {
        content: "+";
      }
      .fa-foo::before {
        content: "+";
      }
      .pf-plus::before {
        content: "+";
      }
      .fa-foo {}
      .should-not-be-picked {}`;
    document.head.appendChild(styleTag);
  });

  afterAll(() => {
    document.removeChild(styleTag);
  });

  it('should render correctly', () => {
    const wrapper = shallow(<IconPicker {...intialProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should cancel correctly', () => {
    const wrapper = mount(<IconPicker
      {...intialProps}
    />);

    expect(wrapper.find(Modal).instance().props.show).toEqual(false);

    wrapper.find('button').at(1).simulate('click');
    const button = wrapper.find('button').last();
    expect(wrapper.find(Modal).instance().props.show).toEqual(true);
    button.simulate('click');
    expect(wrapper.find(Modal).instance().props.show).toEqual(false);
  });

  it('should close correctly', () => {
    const wrapper = mount(<IconPicker
      {...intialProps}
    />);

    expect(wrapper.find(Modal).instance().props.show).toEqual(false);

    wrapper.find('button').at(1).simulate('click');
    const button = wrapper.find('#close-icon-picker-modal');
    expect(wrapper.find(Modal).instance().props.show).toEqual(true);
    button.simulate('click');
    expect(wrapper.find(Modal).instance().props.show).toEqual(false);
  });

  it('should apply correctly', () => {
    const wrapper = mount(<IconPicker
      {...intialProps}
    />);

    expect(wrapper.find(Modal).instance().props.show).toEqual(false);
    const selectedIcon = wrapper.find('#selected-icon');
    expect(selectedIcon.props().className).toEqual('undefined undefined');

    wrapper.find('button').at(1).simulate('click');
    const applyButton = wrapper.find('#apply-icon-picker-icon').first();
    expect(applyButton.instance().props.disabled).toEqual(true);
    wrapper.find(FontIcon).first().simulate('click');
    expect(applyButton.instance().props.disabled).toEqual(false);
    applyButton.simulate('click');
    expect(wrapper.find(Modal).instance().props.show).toEqual(false);
    wrapper.update();
    expect(wrapper.find('#selected-icon').props().className).toEqual('fa fa-crosshairs');
  });

  it('should render tabs correctly', () => {
    const wrapper = mount(<IconPicker
      {...intialProps}
    />);

    expect(wrapper.find(Modal).instance().props.show).toEqual(false);

    wrapper.find('button').at(1).simulate('click');
    expect(wrapper.find(IconsList).at(1).instance().state.icons).toEqual(undefined);
    expect(wrapper.find(IconsList).at(0).instance().state.icons).toEqual([['fa-crosshairs', 'fa-foo']]);
    wrapper.find('a').at(1).simulate('click');
    expect(wrapper.find(IconsList).at(1).instance().state.icons).toEqual([['pf-plus']]);
    wrapper.update();
    expect(wrapper.find(Modal).find('i').props().className).toEqual('pf pf-plus icon-list-item');
  });
});
