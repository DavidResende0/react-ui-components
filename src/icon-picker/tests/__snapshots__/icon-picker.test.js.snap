// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconList test should render correctly 1`] = `
<div
  className="react-ui-components-icon-picker"
>
  <ButtonGroup
    block={false}
    bsClass="btn-group"
    justified={false}
    vertical={false}
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="icon-picker-btn"
      disabled={false}
    >
      <i
        className="undefined undefined"
        id="selected-icon"
        title=".undefined"
      />
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="angle-down"
        type="fa"
      />
    </Button>
  </ButtonGroup>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <button
        aria-hidden="true"
        aria-label="Close"
        className="close"
        id="close-icon-picker-modal"
        onClick={[Function]}
      >
        <Icon
          name="close"
          type="pf"
        />
      </button>
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        Select an icon
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        className="icon-list-modal"
      >
        <Uncontrolled(Tabs)
          activekey={0}
          animation={false}
          id="font-icon-tabs"
          onSelect={[Function]}
        >
          <Tab
            eventKey={0}
            key="fa"
            title="Font Awesome"
          >
            <table
              className="icon-list"
            >
              <tbody>
                <IconsList
                  iconChanged={[Function]}
                  isVisible={true}
                  type="fa"
                />
              </tbody>
            </table>
          </Tab>
          <Tab
            eventKey={1}
            key="pf"
            title="PatternFly"
          >
            <table
              className="icon-list"
            >
              <tbody>
                <IconsList
                  iconChanged={[Function]}
                  isVisible={false}
                  type="pf"
                />
              </tbody>
            </table>
          </Tab>
        </Uncontrolled(Tabs)>
      </div>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        disabled={true}
        id="apply-icon-picker-icon"
        onClick={[Function]}
      >
        Apply
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="btn-cancel"
        disabled={false}
        id="cancel-icon-picker-modal"
        onClick={[Function]}
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
